src/components/TokenEnterAmount.test.tsx

```diff
@@ -0,0 +1,209 @@
+import React from 'react'
+import { getNumberFormatSettings } from 'react-native-localize'
+import { Provider } from 'react-redux'
+import { AmountEnteredIn } from 'src/send/types'
+import { StoredTokenBalance, TokenBalance } from 'src/tokens/slice'
+import { NetworkId } from 'src/transactions/types'
+import { createMockStore } from 'test/utils'
+import {
+  mockCeloTokenBalance,
+  mockCeloTokenId,
+  mockEthTokenId,
+  mockTestTokenTokenId,
+  mockTokenBalances,
+} from 'test/values'
+import TokenEnterAmount, { APPROX_SYMBOL } from './TokenEnterAmount'
+
+jest.mock('react-native-localize')
+
+const getMockStoreTokenBalances = (): Record<string, StoredTokenBalance> => ({
+  ...mockTokenBalances,
+  [mockCeloTokenId]: {
+    ...mockTokenBalances[mockCeloTokenId],
+    priceUsd: '0.5',
+    balance: '10',
+  },
+  [mockEthTokenId]: {
+    ...mockTokenBalances[mockEthTokenId],
+    tokenId: mockEthTokenId,
+    balance: '10',
+    networkId: NetworkId['ethereum-sepolia'],
+    showZeroBalance: true,
+    isNative: true,
+    symbol: 'ETH',
+    priceFetchedAt: Date.now(),
+    name: 'Ether',
+  },
+  [mockTestTokenTokenId]: {
+    // token with no price
+    balance: '10',
+    tokenId: mockTestTokenTokenId,
+    networkId: NetworkId['celo-alfajores'],
+    showZeroBalance: false,
+    isNative: false,
+    symbol: 'TST',
+    name: 'Test Token',
+    decimals: 18,
+    address: '0xtest',
+  },
+})
+const mockStoreTokenBalances = getMockStoreTokenBalances()
+const mockStore = { tokens: { tokenBalances: mockStoreTokenBalances } }
+
+describe('TokenEnterAmount', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest
+      .mocked(getNumberFormatSettings)
+      .mockReturnValue({ decimalSeparator: '.', groupingSeparator: ',' })
+  })
+
+  const mockOnInputChange = jest.fn()
+  const mockToggleAmountType = jest.fn()
+  const mockOnOpenTokenPicker = jest.fn()
+
+  const defaultProps = {
+    token: {
+      ...mockCeloTokenBalance,
+    } satisfies TokenBalance,
+    inputValue: '1',
+    tokenAmount: '1',
+    localAmount: '$0.1',
+    amountType: 'token' as AmountEnteredIn,
+    inputRef: { current: null },
+    onInputChange: mockOnInputChange,
+    toggleAmountType: mockToggleAmountType,
+    onOpenTokenPicker: mockOnOpenTokenPicker,
+    editable: true,
+    testID: 'TokenEnterAmount',
+  }
+
+  it('displays the correct token information', () => {
+    const store = createMockStore(mockStore)
+    const { getByText, getByTestId } = render(
+      <Provider store={store}>
+        <TokenEnterAmount {...defaultProps} />
+      </Provider>
+    )
+    expect(getByTestId('TokenEnterAmount/TokenName')).toHaveTextContent('CELO on Celo Alfajores')
+    expect(getByText('CELO on Celo Alfajores')).toBeTruthy()
+    expect(getByTestId('TokenEnterAmount/SwitchTokens')).toBeTruthy()
+    expect(getByTestId('TokenEnterAmount/TokenSelect')).toBeTruthy()
+    expect(getByTestId('TokenEnterAmount/TokenBalance')).toBeTruthy()
```

Comment:
nit: you can remove this line, since we are asserting on it `toHaveTextContent` next. there might be other places where you can omit the `toBeTruthy()` check too